ipa-core/src/protocol/ipa_prf/boolean_ops/sigmoid.rs

```diff
@@ -158,18 +159,96 @@ where
     ]))
 }
 
+// Sigmoid(
+//     Sum(i = 1..N, neuron(i) in last layer activation times edge weight connecting that neuron to this)
+//   )
+//
+
+// for i in 0..M-1 // For going through all layers
+// for j in 0..N-1 // Current layer
+// for k in 0..N-1 // For previous layer
+// neuron(i*N + j) += neuron((i-1)*N + k) * edge_weight(neuron((i)*N + j), neuron((i-1)*N + k))
+
+// M' neurons wide and here M is M'/N, L layers tall
+pub async fn neural_network<C, S, const M: usize, const N: usize, const MTimesN: usize>(
+    ctx: C,
+    last_layer_neurons: &[BitDecomposed<AdditiveShare<Boolean, N>>; M],
```

Comment:
These are the activations of the last layer of neurons? If so, let's give it a name including that word.